presto-spi/src/main/java/com/facebook/presto/spi/block/RunLengthEncodedBlock.java

```diff
@@ -263,9 +263,15 @@ public String toString()
     }
 
     @Override
-    public void assureLoaded()
+    public Block getLoadedBlock()
     {
-        value.assureLoaded();
+        Block loadedValueBlock = value.getLoadedBlock();
+        if (loadedValueBlock == value) {
+            return this;
+        }
+        else {
+            throw new IllegalArgumentException("Value block is not loaded");
```

Comment:
Why this `throw`? Is it because sources don't produce RLEs?

Reply:
This should be `return new RunLengthEncodedBlock(loadedBlock, positionCount);`